SMN search buttons on SMN CN page

Jakob37 opened the pull request:
This PR adds a functionality or fixes a bug.

Closes #4822 

<img width="405" alt="smn_buttons" src="https://github.com/user-attachments/assets/b50d5812-81b6-4a70-9aba-93e7d3535a8d">

As recommended, I used the buttons on the transcript outliers page as a guide, replacing the HGNC IDs with "SMN1, SMN2".

In the demo instance, the search did not find the "SMN1" and "SMN2" IDs. It did work for a real case with a variant in SMN1 though when trying it out in our test environment. I guess not all HGNC IDs just aren't loaded in the demo.

<img width="411" alt="smn_demo" src="https://github.com/user-attachments/assets/31095143-cbc5-4b0e-9a0c-bb615fb9488d">

A minor detail is the `['']` appearing up to the right. This is the same for the transcript outliers page.

<img width="755" alt="panels_quotes" src="https://github.com/user-attachments/assets/a0bdbb7e-f4db-4697-b9fa-ad3858efb832">

Any feedback is warmly welcome!

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

dnil commented:
Good point with the empty panels indicator. It is not super consistent. But when e.g. searching a case without a set of default panels it will go to "All" which kind of makes sense, but then not revert to none when searching only explicitly a few genes. That is not perfect, and has been the case for a long time. Let's make an issue for it and fix on a quality-of-life prio.

dnil commented:
Feel very free to add the SMN genes to the demo gene set - might be a good exercise. But it is expected so not a problem really.

Jakob37 commented:
> Feel very free to add the SMN genes to the demo gene set - might be a good exercise. But it is expected so not a problem really.

I'll give it a go 👌

Jakob37 commented:
OK, I have added the SMN gene to the resources. It was indeed an informative journey through the Scout resources. Now the demo instance does not complain neither about missing terms nor about not being present in panels.

<img width="707" alt="no_warning_demo" src="https://github.com/user-attachments/assets/32a0f9ea-f569-4b5e-9d70-35d3763f56de">

I have added it to:

* The HGNC terms
* The ensembl gene/transcript/exons for build 37 and 38
* To the test panel

Let me know if I have missed anything!

Otherwise I think it is ready for review @dnil

dnil commented on scout/demo/panel_1.txt:
```diff
@@ -267,3 +267,5 @@
 10542	SBF1						SBF1
 10604	SCO2						SCO2
 713	ARSA						ARSA
+11117	SMN1						SMN1
```
💯

dnil commented on scout/server/blueprints/cases/templates/cases/case_sma.html:
```diff
@@ -50,6 +50,25 @@ <h4 class="mt-3">Case: {{case.display_name}}</h4>
+          <form action="{{url_for('variants.sv_variants', institute_id=institute._id, case_name=case.display_name) }}">
+            <input type="hidden" id="hgnc_symbols" name="hgnc_symbols" value="SMN1, SMN2"></input>
+            <input type="hidden" id="gene_panels" name="gene_panels" value="['']"></input>
+            <button type="submit" class="btn btn-secondary btn-sm" target="_blank" rel="noopener" data-bs-toggle="tooltip" title="SNV and INDEL variants view filtered for the genes SMN1 and SMN2">SVs</button></span>
```
```suggestion
            <button type="submit" class="btn btn-secondary btn-sm" target="_blank" rel="noopener" data-bs-toggle="tooltip" title="Structural variants view filtered for the genes SMN1 and SMN2">SVs</button></span>
```

Jakob37 replied:
Good catch!

dnil commented on tests/commands/view/test_panels_cmd.py:
```diff
@@ -13,7 +13,7 @@ def test_view_panes(mock_app):
     assert result.exit_code == 0
     # a panel should be found
-    assert "panel1\t1.0\t263" in result.output
+    assert "panel1\t1.0\t265" in result.output
```
🥇

dnil reviewed (approved):
Looks good, and thank you for the issue about the current panel display.
